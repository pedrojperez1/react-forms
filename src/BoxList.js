import React, {useState} from "react";
import Box from "./Box";
import NewBoxForm from "./NewBoxForm";
import {v4 as uuid} from "uuid";

const BoxList = ({boxes}) => {
    const [boxList, setBoxList] = useState(boxes);
    const addToBoxList = (newBox) => {
        setBoxList(boxList => [...boxList, newBox]);
    }
    const removeFromBoxList = (id) => {
        setBoxList(boxList => boxList.filter(b => b.id !== id));
    }
    return (
        <div>
            <NewBoxForm addToBoxList={addToBoxList}/>
            {boxList.map(box => (
                <Box
                    id={box.id}
                    backgroundColor={box.backgroundColor} 
                    height={Number(box.height)} 
                    width={Number(box.width)}
                    addToBoxList={addToBoxList}
                    removeFromBoxList={removeFromBoxList}
                />
                )
            )}
        </div>
    )
};

export default BoxList;