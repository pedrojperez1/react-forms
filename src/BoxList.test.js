import { fireEvent, getByPlaceholderText, render } from '@testing-library/react';
import BoxList from './BoxList';

const boxes = [
    // {
    //     id: 123,
    //     backgroundColor: "blue",
    //     height: 200,
    //     width: 250
    // },
    // {
    //     id: 456,
    //     backgroundColor: "salmon",
    //     height: 100,
    //     width: 150
    // }
]

it("should not go up in smoke when rendering", () => {
    render(<BoxList boxes={boxes}/>);
});

it("should render the same each time", () => {
    const {asFragment} = render(<BoxList boxes={boxes}/>);
    expect(asFragment()).toMatchSnapshot();
});

it("should render a new box when form is submitted and delete it when X is pressed", () => {
    const { getByPlaceholderText, getByText, queryAllByText } = render(<BoxList boxes={boxes}/>);
    const backgroundColorInput = getByPlaceholderText("Background color");
    const heightInput = getByPlaceholderText("height");
    const widthInput = getByPlaceholderText("width");
    const submitBtn = getByText("Add");
    fireEvent.change(backgroundColorInput, {target: {value: "green"}});
    fireEvent.change(heightInput, {target: {value: 200}});
    fireEvent.change(widthInput, {target: {value: 200}});
    fireEvent.click(submitBtn);
    expect(queryAllByText("X").length).toBe(1); //first check
    fireEvent.click(getByText("X"));
    expect(queryAllByText("X").length).toBe(0);
});