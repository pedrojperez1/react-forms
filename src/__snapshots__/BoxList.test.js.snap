// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the same each time 1`] = `
<DocumentFragment>
  <div>
    <div
      class="NewBoxForm"
    >
      <form>
        <input
          class="NewBoxForm-input"
          name="backgroundColor"
          placeholder="Background color"
          type="text"
          value=""
        />
        <input
          class="NewBoxForm-input"
          name="height"
          placeholder="height"
          type="text"
          value=""
        />
        <input
          class="NewBoxForm-input"
          name="width"
          placeholder="width"
          type="text"
          value=""
        />
        <button>
          Add
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
